dvc/dependency/repo.py

```diff
@@ -68,14 +68,14 @@ def save(self):
     def dumpd(self):
         return {self.PARAM_PATH: self.def_path, self.PARAM_REPO: self.def_repo}
 
-    def download(self, to):
+    def download(self, to, jobs=None):
         cache = self.repo.cache.local
 
         with self._make_repo(cache_dir=cache.cache_dir) as repo:
             if self.def_repo.get(self.PARAM_REV_LOCK) is None:
                 self.def_repo[self.PARAM_REV_LOCK] = repo.get_rev()
 
-            _, _, cache_infos = repo.fetch_external([self.def_path])
+            _, _, cache_infos = repo.fetch_external([self.def_path], jobs=jobs)
```

Comment:
It wasn't quite apparent how this was being used. Finally, at the 10th/11th level, I found it using `repo.cloud.pull()`. That's quite deep. :(

Reply:
Same to me, they need refactoring.